### PR TITLE
Feature/carousel component

### PR DESCRIPTION
**Pull request by TimDirks**
Adds the bootstrap [Carousel](https://getbootstrap.com/docs/5.1/components/carousel/) component

### CODE REVIEW

File: src/components/Carousel/Carousel.test.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    it('sets active index from slides', async() => {
+        const wrapper = await shallowMount(
+            Carousel,
+            {props},
+        );
+
+        const slides = wrapper.findAll('.carousel-item');
+
+        expect(slides[1].classes()).toContain('active');
```
Comment (barisbora) [inline src/components/Carousel/Carousel.test.ts:67]: Also good to check rest of items does not have `active` class

File: src/components/Carousel/Carousel.vue [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    'Carousel',
+    carousel,
+);
+
+const {classes} = useClasses(computed(() => [
+    props.dark ? 'carousel-dark' : null,
+    props.fade ? 'carousel-fade' : null,
+]));
+
+defineExpose({bsCarousel});
```
Comment (barisbora) [inline src/components/Carousel/Carousel.vue:161]:
```suggestion
defineExpose({bsInstance});
```

File: src/components/Carousel/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import Carousel from './Carousel.vue';
```
Comment (barisbora) [inline src/components/Carousel/index.ts:1]:
```suggestion
import Carousel from '@/components/Carousel.vue';
```

**Review by barisbora — APPROVED** [review]

File: src/components/Carousel/Carousel.vue [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                :slides="slides"
+            >
+                <div
+                    v-for="(slide, index) in slides"
+                    :key="`slide-${index}`"
+                    class="carousel-item"
+                    :class="{active: index === activeIndex}"
+                    :data-bs-interval="slide.interval"
+                >
+                    <slot :slide="slide">
```
Comment (thomasowow) [inline src/components/Carousel/Carousel.vue:22]:
```suggestion
                    <slot
                        :index="index"
                        :slide="slide"
                    >
```

**Review by thomasowow — APPROVED** [review]
Export in [components/index.ts](https://github.com/owowagency/bootstrap-vue/blob/master/src/components/index.ts)

File: src/components/Carousel/index.ts (hunk 3 above)
Comment (TimDirks) [reply src/components/Carousel/index.ts:1]: Relative import is used for all other components too so leaving as is

**Pull request merged**